trade-finance-manager-ui/server/controllers/utilisation-reports/utilisation-report-reconciliation-for-report/extract-query-and-session-data.ts

```diff
@@ -1,42 +1,65 @@
-import { asString } from '@ukef/dtfs2-common';
+import { asString, PremiumPaymentsFilters } from '@ukef/dtfs2-common';
 import { SessionData } from 'express-session';
 import { handleRedirectSessionData } from './handle-redirect-session-data';
 import { getIsCheckboxChecked } from './get-is-checkbox-checked';
 import { getSelectedFeeRecordIdsFromQuery } from './get-selected-fee-record-ids-from-query';
 import { validateFacilityIdQuery } from './validate-facility-id-query';
 
 type GetUtilisationReportReconciliationQuery = {
-  facilityIdQuery?: string;
+  premiumPaymentsFacilityId?: string;
   selectedFeeRecordIdsQuery?: string;
 };
 
+/**
+ * Parses premium payments filters from the query parameters.
+ * @param premiumPaymentsFacilityId - The premium payments facility ID query.
+ * @param originalUrl - The original URL of the request.
+ * @returns An object containing the parsed premium payments filters and any
+ * error resulting from validating the facility ID query.
+ */
+const parsePremiumPaymentsFilters = (facilityIdQuery: string | undefined, originalUrl: string) => {
```

Comment:
can this be optional (will need to change the order) but then will not need to be | undefined